schedule/scheduler.go

```diff
@@ -290,6 +291,40 @@ func (s *Scheduler) StopJob(id JobID) error {
 	return nil
 }
 
+func (s *Scheduler) WaitJob(id JobID) (*JobListResult, error) {
+	s.jobsMu.Lock()
+	job, ok := s.jobs[id]
+	if !ok {
+		return nil, xerrors.Errorf("wait job ID: %d not found", id)
+	}
+	s.jobsMu.Unlock()
+
+	// wait on the job to complete
+	<-job.Job.Done()
+
+	// refetch the job to get latest results (EndedAt and Running will have changed)
+	s.jobsMu.Lock()
+	job, ok = s.jobs[id]
+	if !ok {
+		return nil, xerrors.Errorf("wait job ID: %d not found", id)
+	}
+	s.jobsMu.Unlock()
```

Comment:
Same here, make sure jobsMu is unlocked before return

Reply:
done.